docarray/array/storage/redis/getsetdel.py

```diff
@@ -0,0 +1,74 @@
+from typing import Dict
+
+from docarray import Document
+from docarray.array.storage.base.getsetdel import BaseGetSetDelMixin
+from docarray.array.storage.base.helper import Offset2ID
+
+
+class GetSetDelMixin(BaseGetSetDelMixin):
+    """Provide concrete implementation for ``__getitem__``, ``__setitem__``,
+    and ``__delitem__`` for ``DocumentArrayRedis``"""
+
+    def _get_doc_by_id(self, _id: str) -> 'Document':
+        """Concrete implementation of base class' ``_get_doc_by_id``
+
+        :param _id: the id of the document
+        :return: the retrieved document from redis
+        """
+        try:
+            result = self._client.hgetall(self._doc_prefix + _id)
+            doc = Document.from_base64(result[b'blob'])
+            return doc
+        except Exception as ex:
```

Comment:
what exception do we get from Redis, can we make it more specific?

Reply:
I think since we raise it again it is fine like this